### PR TITLE
Scale cf-testd (Part 1)

### PR DESCRIPTION
**Pull request by vpodzime**
A few minor enhancements for *cf-testd* and the related code. At the same time these are the first steps to make *cf-testd* better prepared to be used for testing at scale.

### CODE REVIEW

**Comment by vpodzime** [comment]
jenkins job: https://ci.cfengine.com/job/pr-pipeline/152/

File: cf-testd/cf-testd.c [hunk 1]
```diff
@@ -64,10 +66,12 @@ typedef struct {
...
 static const struct option OPTIONS[] =
 {
+    {"address",   required_argument, 0, 'a'},
     {"debug",     no_argument,       0, 'd'},
     {"file",      required_argument, 0, 'f'},
     {"help",      no_argument,       0, 'h'},
     {"inform",    no_argument,       0, 'I'},
+    {"port",      required_argument, 0, 'p'},
     {"timestamp", no_argument,       0, 'l'},
     {"verbose",   no_argument,       0, 'v'},
```
Comment (craigcomstock) [inline cf-testd/cf-testd.c]: why are address and port required? They weren't before.

Comment (vpodzime) [reply cf-testd/cf-testd.c]: The options are not required. They just require arguments when used. Or am I missing something?

File: cf-serverd/server_transform.c [hunk 2]
```diff
@@ -468,16 +470,12 @@ static void KeepControlPromises(EvalContext *ctx, const Policy *policy, GenericA
             }
             else if (IsControlBody(SERVER_CONTROL_PORT_NUMBER))
             {
-                CFENGINE_PORT = IntFromString(value);
-                strlcpy(CFENGINE_PORT_STR, value, sizeof(CFENGINE_PORT_STR));
-                Log(LOG_LEVEL_VERBOSE, "Setting default port number to %d",
-                    CFENGINE_PORT);
+                bool ret = SetCfenginePort(value);
```
Comment (olehermanse) [inline cf-serverd/server_transform.c]: Does this break backwards compatibility?

File: cf-testd/cf-testd.c (hunk 1 above)
Comment (craigcomstock) [reply cf-testd/cf-testd.c]: @vpodzime right. looks good.

File: libcfnet/client_code.c [hunk 3]
```diff
@@ -100,6 +103,32 @@ void DetermineCfenginePort()
     Log(LOG_LEVEL_VERBOSE, "Default port for cfengine is %d", CFENGINE_PORT);
 }
 
+bool SetCfenginePort(const char *port_str)
+{
+    /* parse and store the new value (use the string representation of the
+     * parsed value because it may potentially be better/nicer/more
+     * canonical) */
+    long int port;
+    int ret = StringToLong(port_str, &port);
```
[the remark this answers is not in the document]
Comment (craigcomstock) [reply libcfnet/client_code.c]: yeah, IntFromString() is pretty fancy. Maybe just use it again? Then again, I wouldn't expect to be able to specify --port 2k and port is a new option so seems fine to me.

File: libcfnet/server_code.c [hunk 4]
```diff
@@ -70,14 +70,12 @@ static int OpenReceiverChannel(void)
...
+        query.ai_flags |= AI_NUMERICHOST;
     }
 
-    char servname[PRINTSIZE(CFENGINE_PORT)];
-    xsnprintf(servname, sizeof(servname), "%d", CFENGINE_PORT);
-
     /* Resolve listening interface. */
     int gres;
-    if ((gres = getaddrinfo(ptr, servname, &query, &response)) != 0)
+    if ((gres = getaddrinfo(ptr, CFENGINE_PORT_STR, &query, &response)) != 0)
```
[the remark this answers is not in the document]
Comment (vpodzime) [reply libcfnet/server_code.c]: Good catch, this is a good opportunity to make this nicer.

File: libcfnet/client_code.c (hunk 3 above)
[the remark this answers is not in the document]
Comment (vpodzime) [reply libcfnet/client_code.c]: Yeap.

**Pull request merged**